emop-maven-plugin/src/main/java/edu/cornell/VmsMojo.java

```diff
@@ -0,0 +1,277 @@
+     */
+    private void removeDuplicateViolations() {
+        Set<Violation> violationsToRemove = new HashSet<>();
+        for (Violation newViolation : newViolations) {
+            Set<Violation> relevantOldViolations = oldViolations.stream()
+                    .filter(oldViolation -> oldViolation.getSpecification().equals(newViolation.getSpecification()))
+                    .filter(oldViolation -> oldViolation.getClassInfo().equals(newViolation.getClassInfo())
+                                         || isRenamed(oldViolation.getClassInfo(), newViolation.getClassInfo()))
+                    .filter(oldViolation -> hasSameLineNumber(oldViolation.getClassInfo(), oldViolation.getLineNum(),
+                            newViolation.getLineNum()))
+                    .collect(Collectors.toSet());
+
+            if (!relevantOldViolations.isEmpty()) {
+                violationsToRemove.add(newViolation);
+            }
+        }
+        for (Violation violationToRemove : violationsToRemove) {
+            newViolations.remove(violationToRemove);
+        }
+    }
+
+    /**
+     * Determines whether an old class has been renamed to the new one or not.
+     *
+     * @param oldClass Previous possible name of a class
+     * @param newClass Rename being considered
+     * @return Whether the old class name was renamed to the new one
+     */
+    private boolean isRenamed(String oldClass, String newClass) {
+        for (String renamedClass : renames.keySet()) {
+            if (renamedClass.contains(newClass) && renames.get(renamedClass).contains(oldClass)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Determines whether an old line in a file can be mapped to the new line. There's room for optimization.
+     *
+     * @param classInfo Particular class being considered (if the class was renamed, this is the old name)
+     * @param oldLine Original line number
+     * @param newLine New line number
+     * @return Whether the original line number can be mapped to the new line number in the updated version
+     */
+    private boolean hasSameLineNumber(String classInfo, int oldLine, int newLine) {
+        for (String className : lineChanges.keySet()) {
+            if (className.contains(classInfo)) {
+                int offset = 0;
+                for (Integer originalLine : lineChanges.get(className).keySet()) {
+                    if (originalLine <= oldLine) {
+                        offset += lineChanges.get(className).get(originalLine);
+                    }
+                }
+                if (newLine >= oldLine) { // if lines have been inserted
+                    return offset >= 0 && offset >= newLine - oldLine;
+                } else { // if lines have been removed
+                    return offset <= 0 && offset <= newLine - oldLine;
+                }
+            }
+        }
```

Comment:
I don't understand any of this code because I do not see an example. Can you add one to the comments?